tests/run-pass/c_enums.rs

```diff
@@ -1,3 +1,5 @@
+#![allow(const_err)] // don't warn about truncating casts
```

Comment:
Wait, so hard errors inside constants and const-prop-originating warnings outside constants use the same lint name? Isn't that odd?

Reply:
Ah no, turns out https://github.com/rust-lang/rust/pull/67676 made these overflows err-by-default, actually. Interesting.

Reply:
But to me the lint looks wrong here:
```rust
enum Signed {
    Bar = -42,
    Baz,
    Quux = 100,
}
fn signed() -> [i8; 3] {
    let baz = Signed::Baz; // let-expansion changes the MIR significantly
    [Signed::Bar as i8, baz as i8, Signed::Quux as i8]
}
```
`Signed::Bar` is -42 which is well inside the range of `i8`. So why is there an error shown?